bundles/org.openhab.binding.cp750/src/main/java/org/openhab/binding/cp750/internal/CP750Handler.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cp750.internal;
+
+import static org.openhab.binding.cp750.internal.CP750BindingConstants.*;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.*;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.cybso.cp750.CP750Client;
+import de.cybso.cp750.CP750Field;
+import de.cybso.cp750.CP750InputMode;
+import de.cybso.cp750.CP750Listener;
+
+/**
+ * The {@link CP750Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Roland Tapken - Initial contribution
+ */
+public class CP750Handler extends BaseThingHandler implements CP750Listener {
+
+    private final Logger logger = LoggerFactory.getLogger(CP750Handler.class);
+
+    private @Nullable CP750Configuration config;
+
+    private @Nullable CP750Client client;
+
+    private @Nullable ScheduledFuture scheduleFuture;
+
+    public CP750Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (this.client != null) {
+            try {
+                if (command instanceof RefreshType) {
+                    this.client.refresh();
+                    return;
+                }
+
+                switch (channelUID.getId()) {
+                    case CHANNEL_INPUT -> {
+                        if (command instanceof StringType) {
+                            CP750InputMode mode = CP750InputMode.byValue(((StringType) command).toString());
```

Comment:
You can use java 17 feature
```suggestion
                        if (command instanceof StringType commandAsStringType) {
                            CP750InputMode mode = CP750InputMode.byValue((commandAsStringType.toString());
```